packages/core/src/core/adapters/adapters.ts

```diff
@@ -14,18 +14,26 @@
 * limitations under the License.
 */
 import _ from 'lodash'
-import {
-  ObjectType, AdapterOperations, ElemIdGetter, AdapterOperationsContext, ElemID, InstanceElement,
-} from '@salto-io/adapter-api'
+import { ObjectType, AdapterOperations, ElemIdGetter, AdapterOperationsContext, ElemID, InstanceElement, Adapter } from '@salto-io/adapter-api'
 import { createDefaultInstanceFromType } from '@salto-io/adapter-utils'
 import adapterCreators from './creators'
 
 export const getAdaptersCredentialsTypes = (
   names?: ReadonlyArray<string>
 ): Record<string, ObjectType> => {
-  const relevantAdapterCreators = _.isUndefined(names)
-    ? adapterCreators
-    : _.pick(adapterCreators, names)
+  let relevantAdapterCreators: Record<string, Adapter>
+  if (names === undefined) {
+    relevantAdapterCreators = adapterCreators
+  } else {
+    const [existingAdapters, nonExistingAdapters] = _.partition(
```

Comment:
I think that there is no reason to create another variables here using `_.partition()` as you don't really use `existingAdapters` since it equals to `names` in case there was no errors.
```suggestion
  const nonExistingAdapters = names.filter(name => !Object.keys(adapterCreators).includes(name))
```